solr/core/src/java/org/apache/solr/cluster/placement/plugins/AffinityPlacementFactory.java

```diff
@@ -467,7 +569,7 @@ private void makePlacementDecisions(SolrCollection solrCollection, String shardN
         if (candidateAzEntries == null) {
           // This can happen because not enough nodes for the placement request or already too many nodes with replicas of
           // the shard that can't accept new replicas or not enough nodes with enough free disk space.
-          throw new PlacementException("Not enough nodes to place " + numReplicas + " replica(s) of type " + replicaType +
+          throw new PlacementException("Not enough eligible nodes to place " + numReplicas + " replica(s) of type " + replicaType +
```

Comment:
Can we add in the message if there were `withCollection` constraints used in the placement logic? No details, not even indicating that it's the failure reason, just mentioning it...

Reply:
Not sure if it's useful here or output elsewhere already though.

Reply:
it's already handled in `filterNodesWithCollection`